### ISSUE TITLE
UX: Dashboard build mode

### ISSUE THREAD

**Issue by torkelo** [grafana/grafana#6442]
- [x]  Initial design
- [x] Basics work (Add Panel, Row Options)
- [x] Works with dynamic dashboad rows
- [x] Fix broken unit tests
- [x] Fix white theme
- [x] Add shortcut to dashnav settings dropdown
- [x] Decide on name (Edit mode, Build Mode).
- [ ] Add shortcut for toggle legend, collapse all, expand all rows
- [ ] Document shortcuts
- [ ] A way to teach shortcuts in app?
- [ ] Font change, verify ALL forms look ok (some gf-form-labels need to increased as they wrap now)

**Comment by mattttt** [comment]
Been thinking/playing with this a lot. 

I really like the improvements, but I'm not yet convinced that we should introduce the change wholesale. Instead, I'd like to include it as an "Experimental Mode", and leave the existing dashboard green aliens alone for the first beta.

The goal will be to remove them entirely for 4.0 GA, but we need to get more hands using this before making such a dramatic switch. 

In short, I encourage the existing functionality to remain the same for dashboards (Green Alien menu, CTRL+H, dragging panels, etc), and then to add Build Mode (Experimental) in the Dashnav Settings menu.

Thoughts?

**Comment by mattttt** [comment]
@torkelo and I had a great long chat. Basically, in agreeement that build mode continues to be very promising, but it is not perfect yet, and we dont want to impact existing users in any negative way.

He's going to investigate how it can be introduced as a feature toggle as we continue to test it with users.

**Comment by torkelo** [comment]
@mattttt please test the latest from dash-edit-mode branch,  added back green alien, very ruff version. 

Also changed to add panel view auto collapse after pane is added. 

Also added a bunch of shortcuts: 
`b` -> toggle build mode
`g h` -> go home 
`s s` -> search starred

when hovering over panel `e` -> edit panel,  `v` -> view panel. Works as toggles to  `e` while in edit mode  will leave edit mode. `d` will delete panel. 

Really like these new shortcuts :)

**Comment by mattttt** [comment]
Great, will do that momentarily.

**Comment by mattttt** [comment]
This is going great, overall, very nice work. 

- **really** love the Edit Panel keyboard shortcut.  
- CTRL+H to hide the aliens isnt working for me, but I imagine it's because it hasnt been added back in yet. 
- Tabs no longer work while editing a panel:
![tabs-edit](https://cloud.githubusercontent.com/assets/2886187/19939285/a64b2c64-a0fe-11e6-84d0-7a9e21b6b255.gif)
- I'd like to see the "Add Panel" and "Row Options" on the green alien only open the build mode for the row you're interacting with. Triggering build mode for the whole dashboard feels disorienting. Everything changes and it takes a moment to figure out what I wanted to interact with. 

More to come.

**Comment by torkelo** [comment]
after playing with the reintroduced green row icon I am not happy with it. Feels very inconsistent, code is super complex to support it and local "row edit mode".

So now removed it again and replaced it with a very subtle row collapse toggle icon. 

<img width="334" alt="screen shot 2016-11-03 at 11 27 32" src="https://cloud.githubusercontent.com/assets/10999/19962602/c5e00206-a1b8-11e6-95b3-559f8121b061.png">

Click area is the whole left side of the row.

**Comment by torkelo** [comment]
Ok, so build mode was scrapped, replaced with new slide out row menu.

**Comment by torkelo** [comment]
merged to master
